### PR TITLE
XOR-61 [FIX] DS manager cannot load entries with 250K records.

### PR DESCRIPTION
**Pull request by aahuja-r**
Product areas affected
Fliplet Widget Data Source, interface.js

What does this PR do?
DS manager can load 250k+ entries now

JIRA ticket
[JIRA ticket](https://weboo.atlassian.net/browse/XOR-61)

Result
[Screen Recordings](https://api.media.atlassian.com/file/92a35117-6ed8-4c95-b5c5-cdc4dd3ed5cb/artifact/video_1280.mp4/binary?client=435f5956-a841-4ea9-80ca-d94a2672bba6&collection=&max-age=2592000&token=eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiI0MzVmNTk1Ni1hODQxLTRlYTktODBjYS1kOTRhMjY3MmJiYTYiLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOmMwMDg4MTAxLWRjNWItNDM0Zi04N2IxLTczOTI3Nzk3OTNiZSI6WyJyZWFkIl0sInVybjpmaWxlc3RvcmU6ZmlsZTozNzFmOWIyOC0xYTRhLTQ4ZTYtODdjNi0yN2M0YjI2ZDlmNmQiOlsicmVhZCJdLCJ1cm46ZmlsZXN0b3JlOmZpbGU6OTJhMzUxMTctNmVkOC00Yzk1LWI1YzUtY2RjNGRkM2VkNWNiIjpbInJlYWQiXSwidXJuOmZpbGVzdG9yZTpmaWxlOjU1YWEwY2Q2LTU4YmYtNDA1ZC1hMjg3LTcyYWVkZjVhNmUwZCI6WyJyZWFkIl0sInVybjpmaWxlc3RvcmU6ZmlsZTo4MTlmOGNhMC01M2RkLTQ2MmYtOTI0MS0wMmVhODQ5ODkzMzYiOlsicmVhZCJdLCJ1cm46ZmlsZXN0b3JlOmZpbGU6M2JkM2I4MTktZWNiZC00NzdkLWFlYTgtMzUxZDQ0MjU0MTk1IjpbInJlYWQiXSwidXJuOmZpbGVzdG9yZTpmaWxlOmUyMmUxY2ViLTk2OTItNDkzNy04ZmM3LTEwNTQzYzU3MWMzMiI6WyJyZWFkIl0sInVybjpmaWxlc3RvcmU6ZmlsZTo4NWI3NDMwNC0yYmNkLTRjNmItYmI3Yi04MjIxY2M3YzMwNDMiOlsicmVhZCJdfSwiZXhwIjoxNjU2MzE3NTY4LCJuYmYiOjE2NTYzMTY2Njh9.0okwzTMMO50FZkN18Fvx2ViY2J-x0Onmnjrtz0LNn7Y)

Checklist
None

Testing instructions
None

Deployment instructions
None

Author concerns
None

### CODE REVIEW

**Comment by aahuja-r** [comment]
Hi @squallstar 

I am not able to see any button to add reviewers, Requesting you to please review this PR.
 
Best.

File: js/interface.js [hunk 1]
```diff
@@ -337,15 +337,9 @@ function fetchCurrentDataSourceEntries(entries) {
...
-      // Add an empty object in the beginning for _.extend
-      rows.unshift({});
-
-      var computedColumns = _.keys(_.extend.apply({}, rows.map(function(row) {
-        return row.data;
-      })));
-
-      // Remove the first entry
-      rows.shift();
+      let flatObj1 = {};
```
Comment (anerib) [inline js/interface.js:340]: Can we use self-descriptive name for this variable?  Also, use the uniform variable declaration keywords. E.g if using `var` every ever, please use `var` only.

**Review by anerib — CHANGES_REQUESTED** [review]

**Comment by aahuja-r** [comment]
Hi @squallstar 

I am not able to see any button to add reviewers, Requesting you to please review this PR.

Best.

File: js/interface.js [hunk 2]
```diff
@@ -337,15 +337,9 @@ function fetchCurrentDataSourceEntries(entries) {
...
-
-      var computedColumns = _.keys(_.extend.apply({}, rows.map(function(row) {
-        return row.data;
-      })));
-
-      // Remove the first entry
-      rows.shift();
+      var flattenedColumns = {};
+      rows.map(({data}) => data).forEach(dataItem => (flattenedColumns = {...flattenedColumns, ...dataItem}));
+      var computedColumns = _.keys(flattenedColumns);
```
[the remark this answers is not in the document]
Comment (aahuja-r) [reply js/interface.js:343]: Hi @squallstar 

It is done, please review.

**Review by squallstar — APPROVED** [review]

**Pull request merged**